### PR TITLE
Issue-2567 - Fixing not initialized dialog_render

### PR DESCRIPTION
**Pull request by katridi**
## Description
Fix for issue #2567 

## How to test
Run `test/unittests/skills/test_mycroft_skill.py` to ensure that `test_speak_dialog_render_not_intialized` won't fail

## Contributor license agreement signed?
CLA [ ] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/core/PR-2685))

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2685))

**Comment by krisgesling** [comment]
Hi Katridi, thanks for taking a look at this and submitting a fix.

I see you've added a test to `test_common_query_skill.py` but my understanding of the issue is that it can happen on any type of Skill. Is there a reason for it to be in the CQS tests or should this get moved to the MycroftSkill test cases?

**Comment by katridi** [comment]
No, you're right. I've put it in a wrong place. Thanks for pointing out!

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2685))

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2685))

**Review by krisgesling — DISMISSED** [review]
Thanks for switching that over. Looks good now.

Also confirmed that the test fails without the change, and passes with it applied.

**Comment by krisgesling** [comment]
Am working on the Travis CI issue too, it has built successfully but seems to be an issue in reporting this.

File: mycroft/skills/mycroft_skill/mycroft_skill.py [hunk 1]
```diff
@@ -1134,9 +1134,17 @@ def speak_dialog(self, key, data=None, expect_response=False, wait=False):
...
-                   expect_response, wait, meta={'dialog': key, 'data': data})
+        if self.dialog_renderer:
+            data = data or {}
+            self.speak(
+                self.dialog_renderer.render(key, data),
+                expect_response, wait, meta={'dialog': key, 'data': data}
+            )
+        else:
+            self.log.warning(
+                'dialog_render is None, does the locale/dialog folder exists?'
```
Comment (katridi) [inline mycroft/skills/mycroft_skill/mycroft_skill.py:1145]:
```suggestion
                'dialog_render is None, does the locale/dialog folder exist?'
```

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2685))

File: test/unittests/skills/test_mycroft_skill.py [hunk 2]
```diff
@@ -602,6 +602,13 @@ def test_translate_locations(self):
         # Restore lang to en-us
         s.config_core['lang'] = 'en-us'
 
+    def test_speak_dialog_render_not_intialized(self):
```
Comment (krisgesling) [inline test/unittests/skills/test_mycroft_skill.py:605]: typo on initialized

File: test/unittests/skills/test_mycroft_skill.py [hunk 3]
```diff
@@ -602,6 +602,13 @@ def test_translate_locations(self):
...
+        """Test that non-initialzed dialog_renderer won't raise an error."""
```
Comment (krisgesling) [inline test/unittests/skills/test_mycroft_skill.py:606]: Different typo on initialized

**Review by krisgesling — COMMENTED** [review]
I did not see that extra s before, but since we're at it, there's two tiny typo's here.

File: test/unittests/skills/test_mycroft_skill.py (hunk 2 above)
Comment (katridi) [inline test/unittests/skills/test_mycroft_skill.py:605]:
```suggestion
    def test_speak_dialog_render_not_initialized(self):
```

File: test/unittests/skills/test_mycroft_skill.py (hunk 3 above)
Comment (katridi) [inline test/unittests/skills/test_mycroft_skill.py:606]:
```suggestion
        """Test that non-initialized dialog_renderer won't raise an error."""
```

**Review by krisgesling — APPROVED** [review]

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2685))

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2685))

**Pull request merged**